### ISSUE TITLE
Reproducing your results on validation set

### ISSUE THREAD

**Issue by ivanpp** [yanglei18/Mix-Teaching#14]
First thank you for your fantastic work and the codebase you provided. 
Recently, I've been trying to reproduce your result (Mix-Teaching + MonoFlex) on validation set. 

I did 3 rounds of training, and got a best result of 21.3691 (AP40, moderate Car 3D) on validation set, which is about 0.9 less than your 22.27 baseline. I would like to know if I made some mistake while reproducing your results. Here are my settings:

## Round 1

Followed your suggestion in another issue (https://github.com/yanglei18/Mix-Teaching/issues/5#issuecomment-1703683316). For the first round, I trained 5 MonoFlex models **only for 100 epochs** in the supervised mode. Then I did pseudo-labeling. 

## Round 2 and 3

Starting from round 2, I initialized all the student model from the corresponding best pth from previous round. (like a parallel initialization, as shown in the following figure). 

<img width="808" alt="image" src="https://github.com/yanglei18/Mix-Teaching/assets/23610681/11862b88-23f5-4f1f-ab60-4c7b4077218c">

I trained each model on round 2 and 3 for **300 epochs** in the semi-supervised model.
I finally got the best result of 21.3691 at the round 3, and 19.84 average mAP for 5 models. 

I'm wondering if I did something wrong while reproducing your results. Looking forward to your reply. Thank you in advance.

**Comment by yanglei18** [comment]
Thank you for your attention to our work.
For each training round, we initialize all the student models as follows:

![image](https://github.com/yanglei18/Mix-Teaching/assets/26244081/303ce441-f12b-454d-8880-99016e8ab0b3)

**Comment by ivanpp** [comment]
> Thank you for your attention to our work. For each training round, we initialize all the student models as follows:
> 
> ![image](https://private-user-images.githubusercontent.com/26244081/342052112-303ce441-f12b-454d-8880-99016e8ab0b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkxMzY2NjIsIm5iZiI6MTcxOTEzNjM2MiwicGF0aCI6Ii8yNjI0NDA4MS8zNDIwNTIxMTItMzAzY2U0NDEtZjEyYi00NTRkLTg4ODAtOTkwMTZlOGFiMGIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIzVDA5NTI0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2MmY0N2QxY2MzNmI4MDU2YzMwZWVkMGNlZWI3ZjRmM2E3MjNiYmI0YzE1YjI3MGUzNDc0Mjk4NGZjZTNiMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7fOkTPKUnap515lj6FUu7rXdXqIqZxboGbwCeT0VF3Q)

Thank you for your quick reply. However I still have some confusion, hope you can help me with it. 
From my understanding, you're training 3 rounds, 1 with supervised mode 100 epochs, 2 with ssl mode 300 epochs. 
You choose one best model to initialize **all models** of next round.
But do you initialize with the final pth (model_final.pth) or the best pth (model_moderate_best_soft.pth)?

Thank you again!
